### PR TITLE
Apply code format to Geometry/HGCalGeometry

### PR DESCRIPTION
**Pull request by makortel**
#### PR description:

This PR applies the code format to Geometry/HGCalGeometry package (in preparation to the EventSetup consumes migration #26584)

#### PR validation:

The code compiles.

### CODE REVIEW

**Comment by makortel** [comment]
@cmsbuild, please test

**Comment by kpedro88** [comment]
@bsunanda FYI

**Comment by cvuosalo** [comment]
+1

File: Geometry/HGCalGeometry/plugins/moduleDB.cc [hunk 1]
```diff
@@ -2,149 +2,113 @@
...
-				      ROOT::Math::EulerAngles( tvec[jj+3], tvec[jj+4], tvec[jj+5] ) :
-				      ROOT::Math::EulerAngles());
-    const ROOT::Math::Transform3D rt( ea, tl );
+    const ROOT::Math::Translation3D tl(tvec[jj], tvec[jj + 1], tvec[jj + 2]);
+    const ROOT::Math::EulerAngles ea(6 == nTrParm ? ROOT::Math::EulerAngles(tvec[jj + 3], tvec[jj + 4], tvec[jj + 5])
+                                                  : ROOT::Math::EulerAngles());
+    const ROOT::Math::Transform3D rt(ea, tl);
     double xx, xy, xz, dx, yx, yy, yz, dy, zx, zy, zz, dz;
-    rt.GetComponents(xx,xy,xz,dx,yx,yy,yz,dy,zx,zy,zz,dz) ;
-    tr = Tr3D( CLHEP::HepRep3x3( xx, xy, xz,
```
Comment (ianna) [inline Geometry/HGCalGeometry/plugins/moduleDB.cc]: @makortel and @smuzaffar - IMHO, this is easier to read, then one liner

File: Geometry/HGCalGeometry/src/FastTimeGeometry.cc [hunk 2]
```diff
@@ -12,263 +12,246 @@
...
-      trVector.emplace_back( tt.x());
-      trVector.emplace_back( tt.y());
-      trVector.emplace_back( tt.z());
+      const CLHEP::Hep3Vector tt(tr.getTranslation());
+      trVector.emplace_back(tt.x());
+      trVector.emplace_back(tt.y());
+      trVector.emplace_back(tt.z());
       if (6 == numberOfTransformParms()) {
-	const CLHEP::HepRotation rr( tr.getRotation());
-	const ROOT::Math::Transform3D rtr( rr.xx(), rr.xy(), rr.xz(), tt.x(),
```
Comment (ianna) [inline Geometry/HGCalGeometry/src/FastTimeGeometry.cc]: @makortel and @smuzaffar - I think, for matrices this format is more readable.

File: Geometry/HGCalGeometry/src/FastTimeGeometryLoader.cc [hunk 3]
```diff
@@ -10,123 +10,94 @@
...
-    const CLHEP::HepRep3x3 rotation = (zside > 0) ?
-      CLHEP::HepRep3x3(1,0,0,0,1,0,0,0,1) :
-      CLHEP::HepRep3x3(-1,0,0,0,1,0,0,0,-1);
-    const CLHEP::HepRotation hr ( rotation );
-    const CLHEP::Hep3Vector h3v(0,0,zv);
-    const HepGeom::Transform3D ht3d (hr, h3v);
-    DetId detId = (DetId)(FastTimeDetId(detType,0,0,zside));
+    double zv = zside * (topology.dddConstants().getZPos(detType));
+    const CLHEP::HepRep3x3 rotation =
+        (zside > 0) ? CLHEP::HepRep3x3(1, 0, 0, 0, 1, 0, 0, 0, 1) : CLHEP::HepRep3x3(-1, 0, 0, 0, 1, 0, 0, 0, -1);
```
Comment (ianna) [inline Geometry/HGCalGeometry/src/FastTimeGeometryLoader.cc]: @makortel and @smuzaffar - it seems to me that this line is a bit long. I would keep the options in two lines - they would be easier to compare.

File: Geometry/HGCalGeometry/src/FastTimeGeometryLoader.cc [hunk 4]
```diff
@@ -10,123 +10,94 @@
...
+    const CLHEP::HepRotation hr(rotation);
+    const CLHEP::Hep3Vector h3v(0, 0, zv);
+    const HepGeom::Transform3D ht3d(hr, h3v);
+    DetId detId = (DetId)(FastTimeDetId(detType, 0, 0, zside));
 #ifdef EDM_ML_DEBUG
-    std::cout << "FastTimeGeometryLoader:: transf " << ht3d.getTranslation() 
-	      << " and " << ht3d.getRotation();
+    std::cout << "FastTimeGeometryLoader:: transf " << ht3d.getTranslation() << " and " << ht3d.getRotation();
```
Comment (ianna) [inline Geometry/HGCalGeometry/src/FastTimeGeometryLoader.cc]: the line is a bit long?

File: Geometry/HGCalGeometry/src/FastTimeGeometryLoader.cc [hunk 5]
```diff
@@ -10,123 +10,94 @@
...
+  FastTimeGeometry* geom = new FastTimeGeometry(topology);
+  unsigned int numberOfCells = topology.totalGeomModules();  // both sides
+  int detType = topology.detectorType();
 #ifdef EDM_ML_DEBUG
-  std::cout << "Number of Cells " << numberOfCells <<  " for type " << detType
-	    << " of sub-detector " << topology.subDetector() 
-	    << " Shape parameters " << FastTimeGeometry::k_NumberOfShapes
-	    << ":"  << FastTimeGeometry::k_NumberOfParametersPerShape 
-	    << std::endl;
+  std::cout << "Number of Cells " << numberOfCells << " for type " << detType << " of sub-detector "
```
Comment (ianna) [inline Geometry/HGCalGeometry/src/FastTimeGeometryLoader.cc]: I don't think we gain much by merging the lines here.

File: Geometry/HGCalGeometry/src/HGCalGeometry.cc [hunk 6]
```diff
@@ -23,100 +23,82 @@ typedef std::vector<float> ParmVec;
...
 
-HGCalGeometry::HGCalGeometry( const HGCalTopology& topology_ )
-  : m_topology( topology_ ),
-    m_validGeomIds( topology_.totalGeomModules()),
-    mode_( topology_.geomMode()),
-    m_det( topology_.detector()),
-    m_subdet( topology_.subDetector()),
-    twoBysqrt3_(2.0/std::sqrt(3.0)) {
-  
+HGCalGeometry::HGCalGeometry(const HGCalTopology& topology_)
```
Comment (ianna) [inline Geometry/HGCalGeometry/src/HGCalGeometry.cc]: ```topology_``` should be ```topology```

File: Geometry/HGCalGeometry/src/HGCalGeometry.cc [hunk 7]
```diff
@@ -23,100 +23,82 @@ typedef std::vector<float> ParmVec;
...
 }
 
-void HGCalGeometry::newCell( const GlobalPoint& f1 ,
-			     const GlobalPoint& f2 ,
-			     const GlobalPoint& f3 ,
-			     const CCGFloat*    parm ,
-			     const DetId&       detId ) {
-
+void HGCalGeometry::newCell(
+    const GlobalPoint& f1, const GlobalPoint& f2, const GlobalPoint& f3, const CCGFloat* parm, const DetId& detId) {
```
Comment (ianna) [inline Geometry/HGCalGeometry/src/HGCalGeometry.cc]: the older formatting seems better to me

File: Geometry/HGCalGeometry/src/HGCalGeometry.cc [hunk 8]
```diff
@@ -125,527 +107,478 @@ void HGCalGeometry::newCell( const GlobalPoint& f1 ,
...
-    static const int signf[] = {-1,1,1,-1};
-    for (unsigned int i = 0; i < ncorner-1; ++i) {
-      co[i] = GlobalPoint((r+signr[i]*dr)*cos(fi+signf[i]*dfi),
-			  (r+signr[i]*dr)*sin(fi+signf[i]*dfi),
-			  (v.z()+dz));
+    static const int signr[] = {1, 1, -1, -1};
+    static const int signf[] = {-1, 1, 1, -1};
+    for (unsigned int i = 0; i < ncorner - 1; ++i) {
+      co[i] = GlobalPoint(
+          (r + signr[i] * dr) * cos(fi + signf[i] * dfi), (r + signr[i] * dr) * sin(fi + signf[i] * dfi), (v.z() + dz));
```
Comment (ianna) [inline Geometry/HGCalGeometry/src/HGCalGeometry.cc]: the old formatting was more clear for where x, y, z are

File: Geometry/HGCalGeometry/src/HGCalGeometry.cc [hunk 9]
```diff
@@ -125,527 +107,478 @@ void HGCalGeometry::newCell( const GlobalPoint& f1 ,
...
+                                  << p << " r-limit " << rlimit.first << ":" << rlimit.second;
 #endif
   }
   return idnew;
 }
 
-DetId HGCalGeometry::neighborZ(const DetId& idin, 
-			       const MagneticField* bField, int charge,
-			       const GlobalVector& momentum) const {
+DetId HGCalGeometry::neighborZ(const DetId& idin,
```
Comment (ianna) [inline Geometry/HGCalGeometry/src/HGCalGeometry.cc]: how do you decide which function parameters should be all in one line or each in separate one?

File: Geometry/HGCalGeometry/test/HGCalGeometryCornerTester.cc [hunk 10]
```diff
@@ -20,57 +20,53 @@
...
-  const int            cornerType_;
+  const std::string name_;
+  const int cornerType_;
 };
 
-HGCalGeometryCornerTester::HGCalGeometryCornerTester(const edm::ParameterSet& iC) : 
-  name_(iC.getParameter<std::string>("detector")),
-  cornerType_(iC.getParameter<int>("cornerType")) { }
+HGCalGeometryCornerTester::HGCalGeometryCornerTester(const edm::ParameterSet& iC)
+    : name_(iC.getParameter<std::string>("detector")), cornerType_(iC.getParameter<int>("cornerType")) {}
```
Comment (ianna) [inline Geometry/HGCalGeometry/test/HGCalGeometryCornerTester.cc]: I'd prefer to have each parameter in its own line.

File: Geometry/HGCalGeometry/test/HGCalGeometryTester.cc [hunk 11]
```diff
@@ -23,229 +23,204 @@
...
-  
+
 private:
   void doTest(const HGCalGeometry* geom, ForwardSubdetector subdet);
   void doTestWafer(const HGCalGeometry* geom, DetId::Detector det);
   void doTestScint(const HGCalGeometry* geom, DetId::Detector det);
-  
-  std::string    name;
+
+  std::string name;
```
Comment (ianna) [inline Geometry/HGCalGeometry/test/HGCalGeometryTester.cc]: ```name``` should be either ```m_name``` or ```name_```

File: Geometry/HGCalGeometry/test/HGCalTestRecHitTool.cc [hunk 12]
```diff
@@ -29,132 +29,123 @@
...
     geom_ = pG.product();
-    auto geomEE = ((mode_ == 0) ?
-		   static_cast<const HGCalGeometry*>(geom_->getSubdetectorGeometry(DetId::Forward,ForwardSubdetector::HGCEE)) :
-		   static_cast<const HGCalGeometry*>(geom_->getSubdetectorGeometry(DetId::HGCalEE,ForwardSubdetector::ForwardEmpty)));
-    layerEE_    = (geomEE->topology().dddConstants()).layers(true);
-    auto geomFH = ((mode_ == 0) ?
-		   static_cast<const HGCalGeometry*>(geom_->getSubdetectorGeometry(DetId::Forward,ForwardSubdetector::HGCHEF)) :
-		   static_cast<const HGCalGeometry*>(geom_->getSubdetectorGeometry(DetId::HGCalHSi,ForwardSubdetector::ForwardEmpty)));
-    layerFH_    = (geomFH->topology().dddConstants()).layers(true);
+    auto geomEE = ((mode_ == 0) ? static_cast<const HGCalGeometry*>(
```
Comment (ianna) [inline Geometry/HGCalGeometry/test/HGCalTestRecHitTool.cc]: I think, it was better before

**Review by ianna — COMMENTED** [review]
I looks like it does a good job in general. There are a few places where the old style seems better to me - I left some comments.

**Comment by makortel** [comment]
@ianna My understanding is that the format has been pretty much decided already (and I also think that in many cases the earlier, manual formatting looked more appealing).

On the other hand I realized now that this PR conflicts with #26571, and I'd much rather have that one merged first, so maybe I just close this one for now.

**Comment by smuzaffar** [comment]
@ianna , yes Ianna there was a bug and not all of CMS clang-format customization were applied by `scram b code-format` ( https://github.com/cms-sw/cmssw/pull/26599#pullrequestreview-233559015 ) . I have fixed the build rules (should be available in next IB ).

**Pull request closed without merging**